server/src/main/java/org/elasticsearch/index/analysis/PreConfiguredTokenFilter.java

```diff
@@ -75,6 +81,9 @@ private PreConfiguredTokenFilter(String name, boolean useFilterForMultitermQueri
         super(name, cache);
         this.useFilterForMultitermQueries = useFilterForMultitermQueries;
         this.create = create;
```

Comment:
Rather than doing this here, I think we need to do it in CommonAnalysisPlugin so that we can check the index version.  We should be throwing an IllegalArgumentException if the index created version is greater than 7.0, and issuing a deprecation warning otherwise - you can see the same logic used for the `nGram` filter at `CommonAnalysisPlugin#439`.

This also needs to be done in the WordDelimiterTokenFilterFactory constructor (see `NGramTokenFilterFactory` for an example of deprecated/illegal behaviour based on version)

Reply:
Thanks @romseygeek , if I'm not wrong, we should add the following code to ```WordDelimiterTokenFilterFactory```'s constructor and CommonAnalysisPlugin#472:
```
        if (version.onOrAfter(Version.V_7_0_0_alpha1)) {
            throw new IllegalArgumentException(
                "The [word_delimiter] token filter has been removed. Please change the filter name to [word_delimiter_graph] instead.");
        } else {
            deprecationLogger.deprecatedAndMaybeLog("word_delimiter_deprecation",
                "The [word_delimiter] token filter name is deprecated and will be removed in a future version. "
                    + "Please change the filter name to [word_delimiter_graph] instead.");
        }
```
However, I didn't find out how to check deprecation warning in ```WordDelimiterTokenFilterFactoryTests```, which is not subclass of ```ESTestCase```. Could you help me?  Thanks in advance!